src/AppInstallerRepositoryCore/Microsoft/PredefinedInstalledSourceFactory.cpp

```diff
@@ -0,0 +1,175 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#pragma once
+#include "pch.h"
+#include "Microsoft/PredefinedInstalledSourceFactory.h"
+#include "Microsoft/SQLiteIndex.h"
+#include "Microsoft/SQLiteIndexSource.h"
+
+using namespace std::string_literals;
+using namespace std::string_view_literals;
+
+namespace AppInstaller::Repository::Microsoft
+{
+    namespace
+    {
+        // Populates the index with the ARP entries from the given root.
+        void PopulateIndexFromARP(SQLiteIndex& index, HKEY rootKey)
+        {
+            UNREFERENCED_PARAMETER(index);
+            UNREFERENCED_PARAMETER(rootKey);
+        }
+
+        // Populates the index with the entries from MSIX.
+        void PopulateIndexFromMSIX(SQLiteIndex& index)
+        {
+            using namespace winrt::Windows::ApplicationModel;
+            using namespace winrt::Windows::Management::Deployment;
+
+            PackageManager packageManager;
+            auto packages = packageManager.FindPackagesForUserWithPackageTypes({}, PackageTypes::Main);
+
+            // Reuse the same manifest object, as we will be setting the same values every time.
+            Manifest::Manifest manifest;
+            // Add one installer for storing the package family name.
+            manifest.Installers.emplace_back();
+            // Every package will have the same tags currently.
+            manifest.Tags = { "msix" };
+
+            // Fields in the index but not populated:
+            //  AppMoniker - Not sure what we would put.
+            //  Channel - We don't know this information here.
+            //  Commands - We could open the manifest and look for these eventually.
+            //  Tags - Not sure what else we could put in here.
+            for (const auto& package : packages)
+            {
+                // System packages are part of the OS, and cannot be managed by the user.
+                // Filter them out as there is no point in showing them in a package manager.
+                auto signatureKind = package.SignatureKind();
+                if (signatureKind == PackageSignatureKind::System)
+                {
+                    continue;
+                }
+
+                auto packageId = package.Id();
+                Utility::NormalizedString familyName = Utility::ConvertToUTF8(packageId.FamilyName());
+
+                manifest.Id = familyName;
+                manifest.Name = Utility::ConvertToUTF8(package.DisplayName());

```

Comment:
>Utility::ConvertToUTF8(package.DisplayName()) [](start = 32, length = 45)

This might be interesting for different locales

Reply:
The doc says: This property is only supported for UWP app. If you call this property on an instance of a package that was created by any mechanism other than Package.Current, this property returns an empty string ("").
Should we still use this?

---
In reply to: [507499499](https://github.com/microsoft/winget-cli/pull/598#discussion_r507499499) [](ancestors = 507499499)

Reply:
I can tell you that it does return a string in this case (the expected one), so MSDN is wrong here.

---
In reply to: [507500957](https://github.com/microsoft/winget-cli/pull/598#discussion_r507500957) [](ancestors = 507500957,507499499)

Reply:
And this does in fact (on latest internal selfhost of Windows) give back a localized name:
```
PS C:\git\pkg> wingetdev list store
Name                        Id                                            Version
-------------------------------------------------------------------------------------------
WinGet MSStore Source       Microsoft.Winget.MSStore.Source_8wekyb3d8bbwe 2020.819.2444.404
Microsoft Store 환경 호스트 Microsoft.StorePurchaseApp_8wekyb3d8bbwe      12010.1001.2.0
Microsoft Store             Microsoft.WindowsStore_8wekyb3d8bbwe          12010.1001.2.0
```

---
In reply to: [508136853](https://github.com/microsoft/winget-cli/pull/598#discussion_r508136853) [](ancestors = 508136853,507500957,507499499)